### ISSUE TITLE
Docker build stage in which to inject helper not correctly detected

### ISSUE THREAD

**Issue by arthurlogilab** [devspace-cloud/devspace#1252]
**What happened?**  

Failing to `devspace dev` on multistage Dockerfile

```
$ devspace dev 
[snip]
[fatal]  Error building image: Error during image build: cannot inject restart helper into Dockerfile because neither ENTRYPOINT nor CMD was found.

How to fix this:
- Option A: Define an ENTRYPOINT (or CMD) in your Dockerfile
- Option B: Set `images.*.entrypoint` option in your devspace.yaml
- Option C: If you don't want to inject the restart helper, set `images.*.injectRestartHelper` to false
```

Tried to add ENTRYPOINT / CMD since they are defined in `FROM  image`, but that didn't help. 

Dockerfile looks like this : 

```
FROM python:3.7-alpine AS temp
[snip]

FROM logilab/cubicweb
COPY --from=temp dist/asset-*.tar.gz .
[snip]
```
So copied the `ENTRYPOINT/CMD` from `logilab/cubicweb` which felt wrong, but trial'n'error...

```
FROM python:3.7-alpine AS temp
[snip]

FROM logilab/cubicweb
COPY --from=temp dist/asset-*.tar.gz .
[snip]
ENTRYPOINT ["/entrypoint.sh"]
CMD ["start"]
```

Still had the problem. So commented out some parts (the `temp` stage) and it failed, so I found : 

```
version: v1beta9
images:
  app:
    image: registry.logilab.fr/francearchives/cubicweb-francearchives
    preferSyncOverRebuild: true
    injectRestartHelper: true
    appendDockerfileInstructions:
    - USER root
    build:
      docker:
        options:
          target: temp
```

After removing the `build` section which targets temp, it works. Is this multistage incorrectly detected by the `devspace init` stage ? 

**What did you expect to happen instead?**  

**How can we reproduce the bug?** (as minimally and precisely as possible)  

See above extracts of the `Dockerfile` multistage to use. 

**Local Environment:**  
- DevSpace Version: 1.5.5
- Operating System:  linux 
- Deployment method: n/a

**Kubernetes Cluster:**  
- Cloud Provider: n/a
- Kubernetes Version: n/a

**Anything else we need to know?**  

/kind bug

**Comment by FabianKramm** [comment]
@arthurlogilab thanks for reporting this issue! The error message refers to an ENTRYPOINT or CMD within the build stage temp, you have defined it in the next build stage which does not affect temp anymore. So this would work:
```
FROM python:3.7-alpine AS temp
[snip]

ENTRYPOINT ["/entrypoint.sh"]
CMD ["start"]
FROM logilab/cubicweb
COPY --from=temp dist/asset-*.tar.gz .
[snip]
```

**Comment by arthurlogilab** [comment]
Hi @FabianKramm thanks for a prompt response. 

I'm interested in having the 'restart helper' in the main part of my docker instance, not the `temp` part which is only used at build time. My point is that the `devspace init` points to `temp` when it should not.

**Comment by FabianKramm** [comment]
@arthurlogilab I see, we probably should filter the targets that have no ENTRYPOINT or CMD during the `devspace init` process to avoid such errors in the first place.

**Comment by arthurlogilab** [comment]
From the little I understand of devspace as a new user, it would seem a good solution yest.

**Comment by FabianKramm** [comment]
@arthurlogilab this should now be fixed with v5.6.1
